src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/GeneratedCodeContainer.cs

```diff
@@ -106,7 +106,9 @@ public void SetOutput(RazorCSharpDocument csharpDocument, DefaultDocumentSnapsho
                     return;
                 }
 
-                if (_sourceVersion.HasValue && _sourceVersion == SourceVersion.GetNewerVersion(version))
+                if (_sourceVersion.HasValue &&
+                    _sourceVersion != version &&
```

Comment:
If this is a temporary workaround, could you file an issue to revert it and comment that here?

Reply:
Linked to this in the issue. Not gonna leave a comment in the code though; that'll just rot 😄